library/src/blas2/rocblas_sbmv.hpp

```diff
@@ -0,0 +1,172 @@
+
+        y[tx * incy] = alpha * dotp + beta * y[tx * incy];
+    }
+}
+
+template <typename T, typename U, typename V, typename W>
+rocblas_status rocblas_sbmv_template(rocblas_handle handle,
+                                     rocblas_fill   uplo,
+                                     rocblas_int    n,
+                                     rocblas_int    k,
+                                     const V*       alpha,
+                                     rocblas_stride stride_alpha,
+                                     const U*       A,
+                                     rocblas_int    offseta,
+                                     rocblas_int    lda,
+                                     rocblas_stride strideA,
+                                     const U*       x,
+                                     rocblas_int    offsetx,
+                                     rocblas_int    incx,
+                                     rocblas_stride stridex,
+                                     const V*       beta,
+                                     rocblas_stride stride_beta,
+                                     W*             y,
+                                     rocblas_int    offsety,
+                                     rocblas_int    incy,
+                                     rocblas_stride stridey,
+                                     rocblas_int    batch_count)
+{
+    //quick return
+    if(!n || !batch_count)
+        return rocblas_status_success;
+
+    hipStream_t rocblas_stream = handle->rocblas_stream;
+
+    // in case of negative inc shift pointer to end of data for negative indexing tid*inc
+    auto shiftx = incx < 0 ? offsetx - ptrdiff_t(incx) * (n - 1) : offsetx;
+    auto shifty = incy < 0 ? offsety - ptrdiff_t(incy) * (n - 1) : offsety;
+
+    static constexpr int sbmv_DIM_Y = 512;
+    rocblas_int          blocks     = (n - 1) / (sbmv_DIM_Y) + 1;
+    dim3                 grid(blocks, batch_count);
+    dim3                 threads(sbmv_DIM_Y);
+
+    if(handle->pointer_mode == rocblas_pointer_mode_device)
+    {
+        hipLaunchKernelGGL(rocblas_sbmv_kernel<T>,
+                           grid,
+                           threads,
+                           0,
+                           rocblas_stream,
+                           uplo,
+                           n,
+                           k,
+                           alpha,
+                           stride_alpha,
+                           A,
+                           offseta,
+                           lda,
+                           strideA,
+                           x,
+                           shiftx,
+                           incx,
+                           stridex,
+                           beta,
+                           stride_beta,
+                           y,
+                           shifty,
+                           incy,
+                           stridey);
+    }
+    else
+    {
+        if(!*alpha && *beta == 1)
```

Comment:
We are ignoring the possibility of stride_alpha/stride_beta != 0 here. That's fine for now, I just think maybe we should remove them as parameters for now? Alternatively this can be moved into `rocblas_sbmv_kernel`.

Reply:
Yes to move them in and slow down the kernel if they won't be used.  Did you remove stride_alpha/beta from others?

Reply:
I don't think I added stride_alpha or stride_beta in any of the recent ones. I believe it was removed from trsm and everything at the time as well.

Reply:
I'll should just delete them to be consistent but will check again with Juan too that it was a blanket decision.